### ISSUE TITLE
Install platform-specific natives into subdirectories

### ISSUE THREAD

**Issue by ctrueden** [scijava/scijava-maven-plugin#19]
Some projects package platform-specific native binaries into JARs with a naming convention like `foo-1.2.3-native-macosx.jar` or `bar-4.5.6-natives-win32.jar`. We need to teach `copy-jars` to put these into specific subdirectories of the application. I propose creating a property for defining a list of mappings for such artifacts: keys are regex strings, values are destination folders. And set some sensible defaults for the patterns we have already seen.

We will need to think carefully about how this might interact with `scijava.app.subDirectory` when it is set to a non-empty value.

**Comment by ctrueden** [comment]
This is implemented in 35aa8130e7bf6d19384d3ef93457c7d075c419f6, expanded upon in 2992176b8c80b8f769c4f1e12c78d433af6803cc, and made more general in d99e14be60ab0fb97f088582af51502f1abd30a2, with some native-classifier-related improvements also in cca73a71d5280aeebfb0bb167fc1310f14ae4c7f. All available in the new scijava-maven-plugin.

TL;DR: I consider this issue to be thoroughly addressed now! :rocket:

**Comment by tinevez** [comment]
🥳
